lms/djangoapps/badges/backends/badgr.py

```diff
@@ -28,9 +34,16 @@ class BadgrBackend(BadgeBackend):
     badges = []
 
     def __init__(self):
-        super().__init__()
-        if not settings.BADGR_API_TOKEN:
-            raise ImproperlyConfigured("BADGR_API_TOKEN not set.")
+        super(BadgrBackend, self).__init__()
+        if None in (settings.BADGR_USERNAME,
+                    settings.BADGR_PASSWORD,
+                    settings.BADGR_TOKENS_CACHE_KEY,
+                    settings.BADGR_ISSUER_SLUG,
+                    settings.BADGR_BASE_URL):
+            raise ImproperlyConfigured(
+                "One or more of the required settings are not defined. "
+                "Required settings: BADGR_USERNAME, BADGR_PASSWORD, "
+                "BADGR_TOKENS_CACHE_KEY, BADGR_ISSUER_SLUG, BADGR_BASE_URL.")
```

Comment:
Do you perhaps also want to throw `ImproperlyConfigured` in case `BADGR_API_TOKEN`, now obsolete, **is** set?